pkg/build/strategies/sti/sti.go

```diff
@@ -338,12 +365,18 @@ func (builder *STI) createCommandForResultingImage(location string) string {
 }
 
 func (builder *STI) createLabelsForResultingImage() map[string]string {
+	generatedLabels := util.GenerateOutputImageLabels(builder.sourceInfo, builder.config)
+	if len(builder.config.RuntimeImage) > 0 {
+		// don't add labels from the builder image if resulting image is based on another image
+		return generatedLabels
+	}
```

Comment:
looks like no.  I think what should happen here is createLabelsForResultingImage should take an image argument and then the commitBuilder step would pass the builder image arg and the commitRuntime step would pass the runtime image.